### ISSUE TITLE
Proc-macros do not report errors

### ISSUE THREAD

**Issue by hamza1311** [intellij-rust/intellij-rust#9898]
## Environment

* **IntelliJ Rust plugin version:** 0.4.186.5074-223-nightly
* **Rust toolchain version:** 1.65.0 (897e37553 2022-11-02) x86_64-unknown-linux-gnu
* **IDE name and version:** IntelliJ IDEA 2022.3 Ultimate Edition (IU-223.7571.182)
* **Operating system:** Linux 6.0.11-arch1-1
* **Macro expansion:** enabled
* **Additional experimental features:** org.rust.macros.proc.attr, org.rust.cargo.features.settings.gutter, org.rust.cargo.emulate.terminal, org.rust.macros.proc

## Problem description

Errors returned by proc-macros are not reported _at all_.

Usually proc-macros return errors using `compile_error!` macro. The token stream returned also has `Span` information to make the error message more user-friendly. IntelliJ Rust completely ignores these errors and fails silently.

## Steps to reproduce
1. Checkout code from https://github.com/yewstack/yew/pull/2972
2. Use the `yew::html!` macro such that it causes compile error:
```rust
let _ = html! { <> };
```
3. Inspect the expanded code:
```rust
compile_error! {"this opening fragment has no corresponding closing fragment" }
```
4. Notice that there is no error reported.
![image](https://user-images.githubusercontent.com/47357913/207891789-6786a5e6-833c-477c-94ca-53c2c92d6a12.png)

## Additional details

This also results in the IDE assuming that the return type of the macro is `()`, which while technically correct, is most likely not the desired result. Consider the following function:

https://github.com/vlad20012/yew/blob/5355b65ff5f9747cbad801d4b337a5ac7a94d0f4/examples/function_router/src/app.rs#L89-L110

It compiles fine but if `html!` in one of the branches of `match` is invalid, the entire function is highlighted in red:
![image](https://user-images.githubusercontent.com/47357913/207893320-dfd19444-5279-4525-bd8f-1aa683e53753.png)

It is possible to overcome this by returning a default value if macro is invoked by IDE but that is a workaround, not a solution
